Client/Extensions/PaymentRequestExtension.swift

```diff
@@ -0,0 +1,146 @@
+                let body = try JSONDecoder().decode(PaymentRequestBodyParser.self, from: messageData)
+                if body.name == "payment-request-show" {
+                    popup.initPaymentUI()
+                    
+                    guard let detailsData = body.details.data(using: String.Encoding.utf8), let supportedInstrumentsData = body.supportedInstruments.data(using: String.Encoding.utf8) else {
+                        log.error("Error parsing data")
+                        return
+                    }
+                    
+                    let details = try JSONDecoder().decode(PaymentRequestDetailsParser.self, from: detailsData)
+                    
+                    let supportedInstruments =  try JSONDecoder().decode([PaymentRequestSupportedInstrumentsParser].self, from: supportedInstrumentsData)
+                    
+                    guard supportedInstruments.contains(where: {$0.supportedMethods == "bat"}) else {
+                        sendPaymentRequestError(errorName: Strings.notSupportedErrorName, errorMessage: Strings.unsupportedInstrumentMessage)
+                        return
+                    }
+                    
+                    for item in details.displayItems {
+                        popup.addDisplayItemLabel(message: "\(item.label): \(item.amount.value) \(item.amount.currency)\n")
+                        log.info(item.label)
+                    }
+                    
+                    popup.addTotalLabel(message: details.total.label + ":  " + details.total.amount.value + " " + details.total.amount.currency + "\n")
+                    
+                    popup.addButton(title: Strings.paymentRequestPay) { [weak self] in
+                        guard let self = self else {
+                            return .flyDown
+                        }
+                        guard let rewards = self.tab?.rewards, let publisher = self.tab?.publisher, let amount = Double(details.total.amount.value) else {
+                            return .flyDown
+                        }
+                        
+                        rewards.ledger.tipPublisherDirectly(publisher, amount: amount, currency: "BAT") { _ in
+                          // TODO: Handle started tip process
+                            self.response = """
+                                {
+                                  "requestId": "a62c29b3-f840-47cd-b895-4573d3190227",
+                                  "methodName": "bat",
+                                  "details": {
+                                    "transaction_id": "bcbbd947-346d-439f-96b4-101bbd966675",
+                                    "message": "Payment for Ethiopian Coffee!"
+                                  }
+                                }
+                            """
+                            
+                            ensureMainThread {                               
```

Comment:
I don't think this needs to execute on the background thread. This will eventually will just dispatch on background thread via `evaluateJavaScript`.